src/components/app/details/appDetails/SecurityVulnerabilityCard.tsx

```diff
@@ -28,48 +25,57 @@ import LoadingCard from './LoadingCard'
 import { importComponentFromFELibrary } from '../../../common'
 import { useGetAppSecurityDetails } from './AppSecurity'
 
-const isFELibAvailable = importComponentFromFELibrary('isFELibAvailable', null, 'function')
 const SecurityModalSidebar = importComponentFromFELibrary('SecurityModalSidebar', null, 'function')
 
 export const SecurityVulnerabilityCard = ({
     cardLoading,
     appId,
     envId,
     installedAppId,
-    artifactId,
-    isExternalCI,
 }: SecurityVulnerabilityCardType) => {
     const [showScanDetailsModal, setShowScanDetailsModal] = useState<boolean>(false)
-    const isSecurityScanV2Enabled = window._env_.ENABLE_RESOURCE_SCAN_V2 && !!isFELibAvailable
 
     const handleModalClose = () => setShowScanDetailsModal(false)
 
-    const { scanDetailsLoading, scanDetailsResponse, scanDetailsError, reloadScanDetails, severityCount } =
-        useGetAppSecurityDetails({ appId: +appId, envId: +envId, installedAppId, artifactId, isSecurityScanV2Enabled })
+    const { scanResultLoading, scanResultResponse, scanResultError, reloadScanResult } = useGetAppSecurityDetails({
+        appId: +appId,
+        envId: +envId,
+        installedAppId,
+    })
 
-    // Since we get status in v2 Api only
-    const imageScanFailed: boolean = isSecurityScanV2Enabled && scanDetailsResponse?.result.codeScan.status === 'Failed'
-    const imageScanProgressing: boolean =
-        isSecurityScanV2Enabled && scanDetailsResponse?.result.codeScan.status === 'Progressing'
+    if (cardLoading || scanResultLoading) {
+        return <LoadingCard />
+    }
+
+    if (scanResultResponse && (!scanResultResponse.result.scanned || !scanResultResponse.result.isImageScanEnabled)) {
+        return null
+    }
+
+    const { severityCount, totalCount } = getScanToolAndSeverityCount(scanResultResponse?.result)
+
+    const imageScanVulnerabilities = scanResultResponse?.result.imageScan?.vulnerability
+    const imageScanList = imageScanVulnerabilities?.list ?? []
+    const imageScanProgressing: boolean = imageScanList.some(
+        (image) => image.status === 'Running' || image.status === 'Progressing',
+    )
+    const imageScanFailed: boolean = imageScanList[0]?.status === 'Failed'
```

Comment:
Are we sure we will have at least one image?

Reply:
if not, failed will be false in that case